{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "MEMBER",
    "body": "### Feature already in Vim?\n\nN/A\n\n### Feature description\n\nThis is something between (regression) bug or feature request. \r\n\r\n## python remote plugins\r\n\r\nWhen remote (python) plugin throws an error or exception *during an execution of pynvim(neovim) API*, there is no much debugging information available.\r\n\r\nFor example,\r\n\r\n```\r\npython3-rplugin-host: Async request caused an error:\r\nInvalid buffer id: 421\r\nPress ENTER or type command to continue\r\n```\r\n\r\nwithout information about (i) such a neovim API (e.g., `vim.async_call`) is called by which remote plugin, nor (ii) stacktrace of the plugin.\r\n\r\nNote that when a python exception is raised (from the python remote plugin), it correctly prints the stack trace:\r\n```\r\nerror caught in async handler '.../vim/plugged/some-plugin/rplugin/python3/some_module:function:SomeCommand [args]'\r\nTraceback (most recent call last):\r\n  ...\r\n```\r\n\r\n## Lua plugins\r\n\r\nWhen it comes to a lua plugin, sometimes the whole stacktrace are printed in a single line:\r\n\r\nFor example (this bug is not specific to fidget.nvim; it's just one example),\r\n\r\n```\r\nError detected while processing function clever_f#find_with[44]..<SNR>175_getchar:                                                                     \r\nline    2:\r\nError executing vim.schedule lua callback: /Users/wookayin/.vim/plugged/fidget.nvim/lua/fidget.lua:190: E523: Not allowed here^@stack traceback:^@^I[C]: in func\r\ntion 'nvim_open_win'^@^I/Users/wookayin/.vim/plugged/fidget.nvim/lua/fidget.lua:190: in function 'show'^@^I/Users/wookayin/.vim/plugged/fidget.nvim/lua/fidget.l\r\nua:55: in function </Users/wookayin/.vim/plugged/fidget.nvim/lua/fidget.lua:51>\r\n```\r\n\r\nwhere the stacktrace message is printed without actual carriage return but a literal character `^@^I`. I remember this was not the case at some earlier neovim versions (like v0.5), but since some point neovim started to print the stacktrace in a difficult-to-read format.",
    "closed_at": null,
    "closed_by": null,
    "comment_data": [
        {
            "author_association": "CONTRIBUTOR",
            "body": "> where the stacktrace message is printed without actual carriage return but a literal character ^@^I. I remember this was not the case at some earlier neovim versions (like v0.5), but since some point neovim started to print the stacktrace in a difficult-to-read format.\r\n\r\nPre-0.5 there was no detailed stack trace\r\n",
            "created_at": "2022-04-11T19:00:06Z",
            "html_url": "https://github.com/neovim/neovim/issues/18082#issuecomment-1095441115",
            "id": 1095441115,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/18082",
            "node_id": "IC_kwDOAPphoM5BSxrb",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1095441115/reactions"
            },
            "updated_at": "2022-04-11T19:00:06Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1095441115",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/13316262?v=4",
                "events_url": "https://api.github.com/users/mjlbach/events{/privacy}",
                "followers_url": "https://api.github.com/users/mjlbach/followers",
                "following_url": "https://api.github.com/users/mjlbach/following{/other_user}",
                "gists_url": "https://api.github.com/users/mjlbach/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/mjlbach",
                "id": 13316262,
                "login": "mjlbach",
                "node_id": "MDQ6VXNlcjEzMzE2MjYy",
                "organizations_url": "https://api.github.com/users/mjlbach/orgs",
                "received_events_url": "https://api.github.com/users/mjlbach/received_events",
                "repos_url": "https://api.github.com/users/mjlbach/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/mjlbach/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/mjlbach/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/mjlbach",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "An example with\n```\nNVIM v0.11.0-dev                                                                                    \nBuild type: Debug                                                                                   \nLuaJIT 2.1.1727870382                                                                               \n```\nSource the following and hit any key\n```\nlocal function F2()\n    error(\"in F2()\")\nend\nlocal function F1()\n    F2()\nend\nlocal function handler()\n    F1()\nend\nvim.on_key(handler)\n```\noutput for `stack traceback` has no stack for the callback.\n```\nError executing  vim.on_key Lua callback: vim/_editor.lua:0: Error executing 'on_key' with ns_ids '14'\n    Messages: [string \":source (no file)\"]:2: in F2()\nstack traceback:\n        [C]: in function 'error'\n        vim/_editor.lua: in function <vim/_editor.lua:0>\n```\n\n-----\n@zeertzjq @justinmk This is labeled `enhancement`. Isn't not getting a stack trace a bug?",
            "created_at": "2024-10-31T18:32:28Z",
            "html_url": "https://github.com/neovim/neovim/issues/18082#issuecomment-2450569271",
            "id": 2450569271,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/18082",
            "node_id": "IC_kwDOAPphoM6SELg3",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2450569271/reactions"
            },
            "updated_at": "2024-11-01T02:12:50Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2450569271",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/20450427?v=4",
                "events_url": "https://api.github.com/users/errael/events{/privacy}",
                "followers_url": "https://api.github.com/users/errael/followers",
                "following_url": "https://api.github.com/users/errael/following{/other_user}",
                "gists_url": "https://api.github.com/users/errael/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/errael",
                "id": 20450427,
                "login": "errael",
                "node_id": "MDQ6VXNlcjIwNDUwNDI3",
                "organizations_url": "https://api.github.com/users/errael/orgs",
                "received_events_url": "https://api.github.com/users/errael/received_events",
                "repos_url": "https://api.github.com/users/errael/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/errael/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/errael/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/errael",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "`vim.on_key()` is completely unrelated to this issue. Please create a new issue.",
            "created_at": "2024-11-01T02:09:38Z",
            "html_url": "https://github.com/neovim/neovim/issues/18082#issuecomment-2451160555",
            "id": 2451160555,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/18082",
            "node_id": "IC_kwDOAPphoM6SGb3r",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2451160555/reactions"
            },
            "updated_at": "2024-11-01T02:12:55Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2451160555",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/35768171?v=4",
                "events_url": "https://api.github.com/users/zeertzjq/events{/privacy}",
                "followers_url": "https://api.github.com/users/zeertzjq/followers",
                "following_url": "https://api.github.com/users/zeertzjq/following{/other_user}",
                "gists_url": "https://api.github.com/users/zeertzjq/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/zeertzjq",
                "id": 35768171,
                "login": "zeertzjq",
                "node_id": "MDQ6VXNlcjM1NzY4MTcx",
                "organizations_url": "https://api.github.com/users/zeertzjq/orgs",
                "received_events_url": "https://api.github.com/users/zeertzjq/received_events",
                "repos_url": "https://api.github.com/users/zeertzjq/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/zeertzjq/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/zeertzjq/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/zeertzjq",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "> `vim.on_key()` is completely unrelated to this issue. Please create a new issue.\n\nDONE: #31019. (guess I misinterpreted \"... or lua call\")",
            "created_at": "2024-11-01T03:12:37Z",
            "html_url": "https://github.com/neovim/neovim/issues/18082#issuecomment-2451212681",
            "id": 2451212681,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/18082",
            "node_id": "IC_kwDOAPphoM6SGomJ",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2451212681/reactions"
            },
            "updated_at": "2024-11-01T03:12:44Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2451212681",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/20450427?v=4",
                "events_url": "https://api.github.com/users/errael/events{/privacy}",
                "followers_url": "https://api.github.com/users/errael/followers",
                "following_url": "https://api.github.com/users/errael/following{/other_user}",
                "gists_url": "https://api.github.com/users/errael/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/errael",
                "id": 20450427,
                "login": "errael",
                "node_id": "MDQ6VXNlcjIwNDUwNDI3",
                "organizations_url": "https://api.github.com/users/errael/orgs",
                "received_events_url": "https://api.github.com/users/errael/received_events",
                "repos_url": "https://api.github.com/users/errael/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/errael/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/errael/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/errael",
                "user_view_type": "public"
            }
        }
    ],
    "comments": 4,
    "comments_url": "https://api.github.com/repos/neovim/neovim/issues/18082/comments",
    "created_at": "2022-04-11T16:48:49Z",
    "events_url": "https://api.github.com/repos/neovim/neovim/issues/18082/events",
    "html_url": "https://github.com/neovim/neovim/issues/18082",
    "id": 1200247530,
    "labels": [
        {
            "color": "c2e0c6",
            "default": true,
            "description": "feature request",
            "id": 77997476,
            "name": "enhancement",
            "node_id": "MDU6TGFiZWw3Nzk5NzQ3Ng==",
            "url": "https://api.github.com/repos/neovim/neovim/labels/enhancement"
        },
        {
            "color": "c5def5",
            "default": false,
            "description": "",
            "id": 1548913774,
            "name": "logging",
            "node_id": "MDU6TGFiZWwxNTQ4OTEzNzc0",
            "url": "https://api.github.com/repos/neovim/neovim/labels/logging"
        }
    ],
    "labels_url": "https://api.github.com/repos/neovim/neovim/issues/18082/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "I_kwDOAPphoM5HilLq",
    "number": 18082,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 1,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 1,
        "url": "https://api.github.com/repos/neovim/neovim/issues/18082/reactions"
    },
    "repository_url": "https://api.github.com/repos/neovim/neovim",
    "state": "open",
    "state_reason": null,
    "sub_issues_summary": {
        "completed": 0,
        "percent_completed": 0,
        "total": 0
    },
    "timeline_url": "https://api.github.com/repos/neovim/neovim/issues/18082/timeline",
    "title": "More informative message and stacktrace for errors from remote plugins or lua call",
    "updated_at": "2025-01-17T15:59:13Z",
    "url": "https://api.github.com/repos/neovim/neovim/issues/18082",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/1009873?v=4",
        "events_url": "https://api.github.com/users/wookayin/events{/privacy}",
        "followers_url": "https://api.github.com/users/wookayin/followers",
        "following_url": "https://api.github.com/users/wookayin/following{/other_user}",
        "gists_url": "https://api.github.com/users/wookayin/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/wookayin",
        "id": 1009873,
        "login": "wookayin",
        "node_id": "MDQ6VXNlcjEwMDk4NzM=",
        "organizations_url": "https://api.github.com/users/wookayin/orgs",
        "received_events_url": "https://api.github.com/users/wookayin/received_events",
        "repos_url": "https://api.github.com/users/wookayin/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/wookayin/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/wookayin/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/wookayin",
        "user_view_type": "public"
    }
}